src/coreclr/nativeaot/Runtime/unix/PalRedhawkInline.h

```diff
@@ -113,3 +113,14 @@ FORCEINLINE void PalSetLastError(int32_t error)
 {
     errno = error;
 }
+
+FORCEINLINE int32_t PalOsPageSize()
+{
+#ifdef HOST_APPLE
+    return 0x4000;
```

Comment:
This is true for arm64. Is it true on x64 machines as well? Or are we pessimizing it here for simplicity?

Reply:
> Is it true on x64 machines as well?

No, it is not. macOS x64 still uses 4Kb pages.

Reply:
I forgot that OSX on x64 is still possible :-). Yes, on x64 everything uses 4K as OS page size.

Reply:
Is it the case even with x86 emulator on OSX arm64? 

The OS can emulate larger page size than the hardware supports, but doing it the other way around is hard.

Reply:
> Is it the case even with x86 emulator on OSX arm64?

good question. arm64 should support 4K pages. However, what Rosetta does when running on M1 - not sure. I'd guess 16K pages on x64 could be a breaking change. There is one way to find out.

Reply:
> Is it the case even with x86 emulator on OSX arm64?

Yes, is it (with the exception of the first ARM Dev Kits that are no longer in circulation).

Reply:
> Yes, is it (with the exception of the first ARM Dev Kits that are no longer in circulation).

I assume that is the "yes" for 4K pages?  :-)   

I wonder if there are any docs, but emulation could be too much of a corner case and a temporary solution, to document such details.

Reply:
Yes, always 4K pages even in Rosetta emulation.